src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -429,6 +431,78 @@ GetShardCost(uint64 shardId, void *voidContext)
 }
 
 
+Datum
+citus_partitioned_shard_total_size(PG_FUNCTION_ARGS)
+{
+	char *shardName = text_to_cstring(PG_GETARG_TEXT_P(0));
+	uint64 size = 0;
+	const int paramCount = 1;
+	Oid paramTypes[1] = { TEXTOID };
+	Datum paramValues[1] = { CStringGetTextDatum(shardName) };
+
+	char *queryText = "WITH RECURSIVE tables AS ("
```

Comment:
One suggestion I had was to use `pg_partition_root` function instead of this complex query. Can we have another alternatives? Wanted to explicitly note here in case someone else can suggest alternatives